src/macros.rs

```diff
@@ -715,6 +715,82 @@ macro_rules! query_file_scalar_unchecked (
+/// governed by a `cfg`-flag.
+///
+/// ##### Stable Rust: Cargo Build Script
+/// The only solution on stable Rust right now is to create a Cargo build script in your project
+/// and have it print `cargo:rerun-if-changed=migrations`:
+///
+/// `build.rs`
+/// ```
+/// fn main() {
+///     println!("cargo:rerun-if-changed=migrations");
+/// }
+/// ```
+///
+/// You can also do this automatically by running `sqlx migrate build-script`.
+///
+/// See: [The Cargo Book: 3.8 Build Scripts; Outputs of the Build Script](https://doc.rust-lang.org/stable/cargo/reference/build-scripts.html#outputs-of-the-build-script)
+///
+/// #### Nightly Rust: `cfg` Flag
+/// The `migrate!()` macro also listens to `--cfg procmacro2_semver_exempt`, which will enable
+/// the `track_path` feature to directly tell the compiler to watch the `migrations/` directory:
+///
+/// ```sh,ignore
+/// $ env RUSTFLAGS='--cfg procmacro2_semver_exempt' cargo build
+/// ```
+///
+/// Note that this unfortunately will trigger a fully recompile of your dependency tree, at least
+/// for the first time you use it.
+///
+/// You can also set it in `build.rustflags` in `.cargo/config.toml`:
+/// ```toml,ignore
+/// [build]
+/// rustflags = ["--cfg procmacro2_semver_exempt"]
+/// ```
+///
+/// And then continue building and running your project normally.
+///
+/// If you're building on nightly anyways, it would be extremely helpful to help us test
+/// this feature and find any bugs in it.
+///
+/// Subscribe to [the `track_path` tracking issue](https://github.com/rust-lang/rust/issues/73921)
+/// for discussion and the future stabilization of this feature.
+///
+/// **Why not a feature flag?** Feature flags enabling nightly features are technically an anti-pattern
+/// because they take away agency from the person compiling the final binary, even though
+/// the choice of stable vs nightly _should_ be up to them. If you're using SQLx in a library, you
+/// should not set this flag but instead direct your end-users to use it, if applicable.
+///
+/// Additionally, enabling the `nightly` feature would subject you to possible breaking changes
+/// between nightly versions of Rust, even though the version of SQLx hasn't changed. By
+/// forcing the use of the `cfg`-flag, you are required to acknowledge the potential for
+/// breaking changes.
+///
+/// See: [`proc-macro2` docs / Unstable Features](https://docs.rs/proc-macro2/1.0.27/proc_macro2/#unstable-features).
+///
+/// **Why `procmacro2_semver_exempt`?** This is the cfg-flag that the `proc-macro2` crate uses
+/// to enable nightly-only features, and so it made sense to reuse that as it can enable the
+/// proc-macros to use other unstable features as well. If this is a problem for you, e.g.
+/// it's triggering breaking changes in other proc-macros that you need, open an issue and we can
+/// change it to listen to either this cfg-flag or a SQLx-specific one.
```

Comment:
Please do a SQLx-specific cfg, or better, a Cargo feature. I'd be way more okay with opting into breakage from just the `tracked_env` + `tracked_path` APIs, rather than that plus everything `proc_macro2` happens to use under that cfg (if I were to use this feature at all, which I probably won't).

Reply:
As a compromise, I'm adding a SQLx-specific flag and having the macros enable these unstable features if either is present. The `procmacro2_semver_exempt` flag is basically a superset of this flag, so it made sense to just use that to me, but limiting possible breaking changes is a decently compelling argument.

Reply:
I'm still not seeing the cfg feature's advantage over a cargo feature. But I also don't plan to use this 🤷

Reply:
It's not that there's anything particularly advantageous, I just don't feel like a Cargo feature is the right way to flag this.